yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -121,13 +121,20 @@ private[spark] abstract class YarnSchedulerBackend(
     }
   }
 
+  private[cluster] def prepareRequestExecutors(requestedTotal: Int): RequestExecutors = {
```

Comment:
I certainly could use a mock instead -- but I actually thought that refactoring for testing is preferable, and mocks are only used when doing that kind of refactoring is too complex for the task at hand / code you don't control / etc.  (plus some judgement of which is cleaner).  In this case, I think separating the logic from just sending msgs around seems like a good thing.

but don't have a really strong preference.